components/NewNFTBook.vue

```diff
@@ -956,8 +982,43 @@ async function submitEditedClass () {
   }
 }
 
-function onSubmit () {
-  return isEditMode.value ? submitEditedClass() : submitNewClass()
+async function addNewEdition () {
+  try {
+    isLoading.value = true
+    const p = mapPrices(prices.value)
+    const price = p[0]
+    const autoDeliverCount = p
+      .filter(price => price.isAutoDeliver)
+      .reduce((acc, price) => acc + price.stock, 0)
+
+    let autoDeliverNFTsTxHash
+
+    if (autoDeliverCount > 0) {
+      if (!wallet.value || !signer.value) {
+        await initIfNecessary()
+      }
+      if (!wallet.value || !signer.value) {
+        throw new Error('Unable to connect to wallet')
+      }
+      autoDeliverNFTsTxHash = await sendNFTsToAPIWallet(
+        [classIdInput.value as string],
+        [autoDeliverNftIdInput.value as string],
+        autoDeliverCount,
+        signer.value,
+        wallet.value
+      )
+    }
+    await bookStoreApiStore.addEditionPrice(classId.value as string, newEditionIndex.value as string, {
+      price,
+      autoDeliverNFTsTxHash
+    })
+
+    emit('submit')
+  } catch (error) {
+    console.error(error)
```

Comment:
Add eslint ignore

Reply:
[🎨 Improve condition checks and price declaration](https://github.com/likecoin/nft-book-press/pull/332/commits/91bc387a398467f0a79548cf4bfd12dd77ce4c5f)